src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -272,7 +272,10 @@ public Screen getNextScreen(boolean needsFullEntityScreen, EntityScreenContext e
         } else if (next.equals(SessionFrame.STATE_COMMAND_ID)) {
             MenuScreen menuScreen = new MenuScreen();
             menuScreen.init(sessionWrapper);
-            persistentMenuHelper.addMenusToPeristentMenu(menuScreen);
+            // if we are not respecting relevancy, we only want to add root menu options to persistent menu
+            if (persistentMenuHelper.getPersistentMenu().isEmpty() || entityScreenContext.isRespectRelevancy()) {
```

Comment:
Don't we want this to always evaluate to `false`  if `entityScreenContext.isRespectRelevancy()` is `false` even if persistent menu is empty?

Reply:
that's because we on the minimum want to show the root menu options in persistent menu (instead of having it as empty) even if entityScreenContext.isRespectRelevancy() is false

Reply:
By root menu, do you mean the menu options on the first screen upon entering an app? Would this potentially add a hidden module or hidden form?

maybe im confused in what isRespectRelevancy is.

Reply:
> do you mean the menu options on the first screen upon entering an app? 
yes

> Would this potentially add a hidden module or hidden form?
not really, but only the visible options. 

> maybe im confused on what isRespectRelevancy represents

it's a setting that can allow user to see hidden menus, currently it's only used for going to a hidden menu with smart links. When it's `false` we only show the visible options on root menu to the user.